datumaro/plugins/open_images_format.py

```diff
@@ -591,6 +723,58 @@ def _save_subsets(self, annotation_writer):
                                     for bool_attr in OpenImagesPath.BBOX_BOOLEAN_ATTRIBUTES
                                 },
                             })
+                        elif annotation.type is AnnotationType.mask:
+                            mask_dir = osp.join(self._save_dir, OpenImagesPath.MASKS_DIR, item.subset)
+
+                            box_id_str = annotation.attributes.get('box_id')
+
+                            if box_id_str:
+                                if _RE_INVALID_PATH_COMPONENT.fullmatch(box_id_str):
+                                    raise UnsupportedBoxIdError(item_id=item.id, box_id=box_id_str)
+                            else:
+                                # find a box ID that isn't used in any other annotations
+                                while True:
+                                    box_id_str = format(next_box_id, "08x")
+                                    next_box_id += 1
+                                    if box_id_str not in existing_box_ids:
```

Comment:
Are the ids unique for the whole dataset?

Reply:
The box IDs are only unique per image, which is why `existing_box_ids` is recreated for each image.

I guess technically they could be unique per image _and_ label, but there are no duplicate (image, box_id) pairs in the whole dataset, so it's not a particularly relevant distinction.